raft/src/test/java/org/apache/kafka/raft/RequestManagerTest.java

```diff
@@ -120,10 +131,17 @@ public void testIgnoreUnexpectedResponse() {
         );
 
         long correlationId = 1;
-        cache.onRequestSent(node, correlationId, time.milliseconds());
-        assertFalse(cache.isReady(node, time.milliseconds()));
-        cache.onResponseResult(node, correlationId + 1, true, time.milliseconds());
-        assertFalse(cache.isReady(node, time.milliseconds()));
+        cache.onRequestSent(node, correlationId, time.milliseconds(), fetch);
+        assertFalse(cache.isReady(node, time.milliseconds(), fetch));
+        cache.onResponseResult(node, correlationId + 1, true, time.milliseconds(), fetch);
+        assertFalse(cache.isReady(node, time.milliseconds(), fetch));
+
+        // completing a request of a different type should not affect the state of other requests
+        cache.onRequestSent(node, correlationId, time.milliseconds(), updateVoter);
+        assertFalse(cache.isReady(node, time.milliseconds(), updateVoter));
+        cache.onResponseResult(node, correlationId, true, time.milliseconds(), updateVoter);
+        assertTrue(cache.isReady(node, time.milliseconds(), updateVoter));
+        assertFalse(cache.isReady(node, time.milliseconds(), fetch));
```

Comment:
Create a new `@Test` method for testing that the request manager can have multiple inflight requests.